opengovsg/plumber
# PLU-7: [MULTI-ROW-9] Create MultiRow component

## Problem
We need a way for users to define multiple rows of values to perform an operation on.

E.g. in vault workspace, update `N` columns by setting `column_i` to `value_i`. In this case, we have `N` rows, where each row consists of 2 values: `column_i` and `value_i`.

This is similar to the multi-select problem, except that users need to configure `N`, as well as values associated with each row in `N`.

## Solution
We will create a "multi-row" component that allows:
1. Apps to define what values to collect in each row
2. Users to add an arbitrary number of rows and specify the values in each row.

https://github.com/opengovsg/plumber/assets/135598754/aa79b27b-6509-443a-84fd-25a9a923a6de

## Tests
- Check that I can add as many rows as I want
- Check that I can delete rows
- Check that when I add rows after deleting, I don't see the old deleted data.
- Check that multi-row inputs translate to an array on the backend, by trying it out with the new Vault Workspace "update multiple columns" action (see #118).

## Deploy Notes
**New dependencies**:
- `react-icons`: For BoxIcons
